### ISSUE TITLE
How can Clients dynamically determine the ID-Field of a collection?

### ISSUE THREAD

**Issue by sebastianfrey** [opengeospatial/ogcapi-features#689]
Most recently, I've been working extensively with the OGC API Features and I noticed that there it is impossible for a web client to dynamically determine the ID column of a collection.

Consider the following scenario:

A web client, which allows to browse OGC API Feature collections in a generic attribute table, needs to re-fetch a specific feature, in order to update the related row.

Without any additional configuration or explicit knowledge about the underlying dataset, that client would not be able to re fetch a feature by its id.

Therefore I would suggest to add a new property to the `Collection` interface. I can imagine a property like `idColumn` or `idField`:

```
{
   "title": "Collection of Features XYZ",
   "id": "xyz",
   "idField": "the_id_field",
   ....
}
```

I dont't know if the `Queryables` interface can be leveraged for this, but as I understand there is no way to mark a certain field as ID within a JSON schema.

**Comment by cportele** [comment]
The behavior depends on the feature encoding. For example, in a GeoJSON encoding the `id` member of the GeoJSON feature object is used. 

So, if you access `https://example.com/collections/buildings/items` with `Accept: application/geo+json` you will get a GeoJSON feature collection. To directly access an individual feature get the  value of the `id` member and fetch `https://example.com/collections/buildings/items/{id}`.

The id may also be encoded in the `properties` object, but this is not required.

For GML it is similar, the `gml:id` attribute is used.

For other encodings it will depend on the specification of the format.

Does that answer your question?

**Comment by sebastianfrey** [comment]
You're right, one might use the get feature by ID API endpoint. But let me ask the question another way:

How can I query multiple features by ID, using the `filter` query parameter, without having explicit knowledge about the name of the ID column.

Example query:

```
${ID_COLUMN} IN (1,2,...,n)
```

Without knowing what the name of `${ID_COLUMN}` is, such querys can not be performed dynamically.

**Comment by cportele** [comment]
This is supported by Part 3, if the API declares a queryable for this.

General question to the group: Would it make sense to add a recommendation to define such a queryable with the name 'ID' (or 'id' or ?)?

**Comment by jvanulde** [comment]
> This is supported by Part 3, if the API declares a queryable for this.
> 
> General question to the group: Would it make sense to add a recommendation to define such a queryable with the name 'ID' (or 'id' or ?)?

Can it not be inferred through the IRI for an item in a given collection?

**Comment by cportele** [comment]
Meeting 2022-05-09: The ability to request features by identifier is a common practice, probably it would be better to have a separate query parameter (e.g. `ids`) like `bbox`, `datetime` to filter on the identifier. It should be a separate conformance class, not as part of Part 3. That is, it should be specified in a new part. The label was changed to "Future Work", proposals are welcome.

**Comment by sebastianfrey** [comment]
@cportele  I see that the proposal has been accepted. I would be interested to know what exactly the next steps are? I am aware that this will take some time, I am just curious how exactly the process works. =)
